embedded_sass.proto

```diff
@@ -4,155 +4,176 @@
+      DEBUG = 2;
+    }
+    Type type = 2;
 
-  // How to format the CSS output.
-  OutputStyle style = 4;
+    // The text of the message.
+    string message = 3;
 
-  // Whether to generate a source map. Note that this will *not* add a source
-  // map comment to the stylesheet; that's up to the embedder or its users.
-  bool source_map = 5;
-}
+    // The span associated with this message. Optional.
+    SourceSpan span = 4;
 
-// An outbound response that contains the result of a compilation.
-message CompileResponse {
-  // The compilation's request id.
-  int32 id = 1;
-
-  // A message indicating that the Sass file was successfully compiled to CSS.
-  message CompileSuccess {
-    // The compiled CSS.
-    string css = 1;
-
-    // The JSON-encoded source map, or the empty string if
-    // `CompileRequest.source_map` was `false`.
-    //
-    // The compiler must not add a `"file"` key to this source map. It's the
-    // embedder's (or the embedder's user's) responsibility to determine how the
-    // generated CSS can be reached from the source map.
-    string source_map = 2;
-  }
-
-  // A message indicating that the Sass file could not be successfully compiled
-  // to CSS.
-  message CompileFailure {
-    // A message describing the reason for the failure.
-    string message = 1;
-
-    // The span associated with the failure. Optional.
-    SourceSpan span = 2;
-
-    // The stack trace associated with the failure.
+    // The stack trace associated with this message.
     //
     // The empty string indicates that no stack trace is available. Otherwise,
     // the format of this stack trace is not specified and is likely to be
     // inconsistent between implementations.
-    string stack_trace = 3;
-  }
-
-  // The success or failure result of the compilation. Mandatory.
-  oneof result {
-    // The result of a successful compilation.
-    CompileSuccess success = 2;
-
-    // The result of a failed compilation.
-    CompileFailure failure = 3;
+    string stack_trace = 5;
   }
-}
 
-// An oubound event indicating that a message should be displayed to the user.
-message LogEvent {
-  // The request id for the compilation that triggered the message.
-  uint32 compilation_id = 1;
-
-  // The type of message.
-  enum Type {
-    // A warning for something other than a deprecated Sass feature. Often
-    // emitted due to a stylesheet using the `@warn` rule.
-    WARNING = 0;
-
-    // A warning indicating that the stylesheet is using a deprecated Sass
-    // feature. Compilers should not add text like "deprecation warning" to
-    // deprecation warnings; it's up to the embedder to determine how to signal
-    // that to the user.
-    DEPRECATION_WARNING = 1;
-
-    // A message generated by the user for their own debugging purposes.
-    DEBUG = 2;
+  // The wrapped message. Mandatory.
+  oneof message {
+    CompileResponse compileResponse = 1;
+    LogEvent logEvent = 2;
```

Comment:
nested LogEvents? I'm fine with them, I'm just curious about their purpose.

Reply:
It's kind of hard to tell here, but this is directly within `OutboundMessage`, not `LogEvent`.

Reply:
and I even double checked, ty